test/distributed/fsdp/test_checkpoint_wrapper.py

```diff
@@ -56,6 +56,58 @@ def test_load_activation_checkpointed_module(self):
         for p1, p2 in zip(lin.parameters(), lin_new.parameters()):
             self.assertEqual(p1, p2)
 
+    def test_checkpoint_wrapper_kwarg_support(self):
+        class MyModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.lin = nn.Linear(10, 10)
+
+            def forward(self, a, b, c=None, d=None, **kwargs):
+                return (
+                    self.lin(a),
+                    self.lin(b),
+                    self.lin(c),
+                    self.lin(d)
+                )
+
+
+        for wrapper in [
+            partial(checkpoint_wrapper, checkpoint_impl=CheckpointImpl.REENTRANT),
+            partial(checkpoint_wrapper, checkpoint_impl=CheckpointImpl.NO_REENTRANT),
+            partial(checkpoint_wrapper, offload_to_cpu=True),
+        ]:
+            with self.subTest(wrapper=wrapper):
+                model = wrapper(MyModel())
+                self.assertTrue(isinstance(model, CheckpointWrapper))
+                # Verify kwargs can be passed in
+                inp = torch.ones(4, 10, requires_grad=True)
+                out = model(inp, inp, c=inp, d=inp, e=inp, f=inp)
+                self.assertTrue(isinstance(out, tuple))
+                self.assertEqual(4, len(out))
+                # Without kwargs should have equivalent gradient requirements.
+                out_no_kwarg = model(inp, inp, inp, inp)
+                for t1, t2 in zip(out_no_kwarg, out):
+                    self.assertEqual(t1, t2)
+                    self.assertEqual(t1.requires_grad, t2.requires_grad)
+
+        # Test model that enforces kwarg inputs
```

Comment:
This just means that the model forward function enforces kwargs, because its signature is:

```
def forward(self, *, a=None, b=None)
```

so all other arguments than `self` have to be kwargs. Test is mostly here because it was a request from FLAVA team to make sure that this works.